IOPool/Output/src/RootOutputFile.cc

```diff
@@ -682,70 +684,53 @@ namespace edm {
 
   void RootOutputFile::fillBranches(
                 BranchType const& branchType,
-                Principal const& principal,
+                OccurrenceForOutput const& occurrence,
                 StoredProductProvenanceVector* productProvenanceVecPtr,
-                ModuleCallingContext const* mcc) {
+                ProductProvenanceRetriever const* provRetriever) {
 
     std::vector<std::unique_ptr<WrapperBase> > dummies;
 
-    bool const fastCloning = (branchType == InEvent) && (whyNotFastClonable_ == FileBlock::CanFastClone);
-
     OutputItemList const& items = om_->selectedOutputItemList()[branchType];
 
-    std::set<StoredProductProvenance> provenanceToKeep;
+    bool const doProvenance = (productProvenanceVecPtr != nullptr) && (om_->dropMetaData() != PoolOutputModule::DropAll);
+    bool const keepProvenanceForPrior = doProvenance && om_->dropMetaData() != PoolOutputModule::DropPrior;
 
-    ProductProvenanceRetriever const* provRetriever = nullptr;
-    {
-      EventPrincipal const* eventPrincipal = dynamic_cast<EventPrincipal const*>(&principal);
-      if (eventPrincipal) {
-        provRetriever =eventPrincipal->productProvenanceRetrieverPtr();
-        assert(provRetriever);
-      }
-    }
-    
+    bool const fastCloning = (branchType == InEvent) && (whyNotFastClonable_ == FileBlock::CanFastClone);
```

Comment:
@Dr15Jones A decision was made a long time ago to never fast copy anything in the run or lumi trees. It makes life a lot simpler.